lib/utils/timetable.ts

```diff
@@ -58,6 +58,17 @@ export const resolveTimetableEventsCSV = (
     }
   });
 
+  const newEvents = JSON.parse(
+    JSON.stringify(
+      timetableEvents.filter(
+        (event) =>
+          !eventsToDownload.some((e) => e.reference === event.reference)
+      )
+    )
+  );
```

Comment:
It's safe to have it here but we probably don't need the deep copy here because we're not doing any mutation here or in objectArrayToCSVString

Reply:
I removed it :)